src/portfft/dispatcher/subgroup_dispatcher.hpp

```diff
@@ -123,7 +123,8 @@ PORTFFT_INLINE void subgroup_impl(const T* input, T* output, const T* input_imag
   Idx fft_size = factor_sg * factor_wi;
   Idx n_cplx_per_sg = n_ffts_per_sg * fft_size;
   Idx n_reals_per_sg = n_ffts_per_sg * n_reals_per_fft;
-  Idx id_of_fft_in_sg = subgroup_local_id / factor_sg;
+  // id_of_fft_in_sg must be < n_ffts_per_sg. Subtraction ensure this for trailing work-items.
+  Idx id_of_fft_in_sg = subgroup_local_id / factor_sg - subgroup_local_id / (factor_sg * n_ffts_per_sg);
```

Comment:
Integer division is relatively expensive on GPUs and could be replaced with a comparison as we know `id_of_fft_in_sg` will be too big by at most 1.
```suggestion
  Idx id_of_fft_in_sg = subgroup_local_id / factor_sg - (subgroup_local_id <= factor_sg * n_ffts_per_sg);
```

Reply:
Done in [84aa2cf](https://github.com/codeplaysoftware/portFFT/pull/130/commits/84aa2cf368a8ccba8d036cbc89e60dec8590e775).